### PR TITLE
meta: add parentKey to track all parents of a file

### PR DESCRIPTION
**Pull request by SandyXSD**
TODO: dump/load

### CODE REVIEW

File: pkg/meta/sql.go [hunk 1]
```diff
@@ -2062,6 +2141,19 @@ func (m *dbMeta) CopyFileRange(ctx Context, fin Ino, offIn uint64, fout Ino, off
 	return errno(err)
 }
 
+func (m *dbMeta) doGetParents(ctx Context, inode Ino) map[Ino]int {
+	var rows []linkParent
+	if err := m.db.Find(&rows, &linkParent{Inode: inode}); err != nil {
```
Comment (davies) [inline pkg/meta/sql.go:2146]: inside roTxn

File: pkg/meta/sql.go [hunk 2]
```diff
@@ -61,6 +61,12 @@ type edge struct {
 	Type   uint8  `xorm:"notnull"`
 }
 
+type linkParent struct {
+	Id     int64 `xorm:"pk bigserial"`
+	Inode  Ino   `xorm:"notnull"`
```
Comment (davies) [inline pkg/meta/sql.go:66]: index ?

File: pkg/meta/sql.go (hunk 1 above)
Comment (SandyXSD) [reply pkg/meta/sql.go:2146]: OK, need merging main branch

File: pkg/meta/sql.go (hunk 2 above)
Comment (SandyXSD) [reply pkg/meta/sql.go:66]: What index? Do you mean count of each parent?

Comment (davies) [reply pkg/meta/sql.go:66]: We need to find all parents for given `inode`, so `inode` should be a index

**Pull request merged**